server/src/main/java/org/elasticsearch/index/query/SpanTermQueryBuilder.java

```diff
@@ -78,7 +83,25 @@ protected SpanQuery doToQuery(SearchExecutionContext context) throws IOException
             term = new Term(fieldName, BytesRefs.toBytesRef(value));
         } else {
             Query termQuery = mapper.termQuery(value, context);
-            term = MappedFieldType.extractTerm(termQuery);
+            List<Term> termsList = new ArrayList<>();
+            termQuery.visit(new QueryVisitor() {
+                @Override
+                public QueryVisitor getSubVisitor(BooleanClause.Occur occur, Query parent) {
+                    if (occur == BooleanClause.Occur.MUST || occur == BooleanClause.Occur.FILTER) {
+                        return this;
+                    }
+                    return EMPTY_VISITOR;
+                }
+
+                @Override
+                public void consumeTerms(Query query, Term... terms) {
+                    termsList.addAll(Arrays.asList(terms));
+                }
+            });
+            if (termsList.size() != 1) {
```

Comment:
So this is because we only didn't error on TermQuery instances which are sure to have only one term, and TIS queries that have on term, in the original method? Just to confirm I understand this right.

Reply:
I added a comment. I came to the conclusion that this is not needed. I don't know how you can get either 0 or more than one term out of mapper.termQuery. But I no longer get why we had a special case for TermInSetQuery there either. How can you get that query again out of mapper.termQuery ? Did you see test fail when we commented that out in the lucene 10 branch?

So far, I've seen one legit build failure, we were hitting the previous "not a term query" error on match_only_text fields, and tests were expecting a 400 error without checking the error message. In reality, a term can totally be extracted out of those queries, but they return a ConstantScoreQuery which we don't unwrap, hence it seems we were lucky to return the expected status code for a situation that down the line would have raised an IllegalStateException. 

I added a specific check above to verity that there are positions and throw a specific error for that case.